internal/controller/postgrescluster/pgadmin.go

```diff
@@ -185,53 +185,58 @@ func (r *Reconciler) reconcilePGAdminDeployment(
 
 	// if the shutdown flag is set, set pgAdmin replicas to 0
 	if cluster.Spec.Shutdown != nil && *cluster.Spec.Shutdown {
-		deploy.Spec.Replicas = initialize.Int32(0)
+		sts.Spec.Replicas = initialize.Int32(0)
 	} else {
-		deploy.Spec.Replicas = cluster.Spec.UserInterface.PGAdmin.Replicas
+		sts.Spec.Replicas = cluster.Spec.UserInterface.PGAdmin.Replicas
 	}
 
 	// Don't clutter the namespace with extra ReplicaSets.
-	deploy.Spec.RevisionHistoryLimit = initialize.Int32(0)
-
-	// Ensure that the number of Ready pods is never less than the specified
-	// Replicas by starting new pods while old pods are still running.
-	// - https://docs.k8s.io/concepts/workloads/controllers/deployment/#rolling-update-deployment
-	deploy.Spec.Strategy.Type = appsv1.RollingUpdateDeploymentStrategyType
-	deploy.Spec.Strategy.RollingUpdate = &appsv1.RollingUpdateDeployment{
-		MaxUnavailable: intstr.ValueOrDefault(nil, intstr.FromInt(0)),
+	sts.Spec.RevisionHistoryLimit = initialize.Int32(0)
+
+	// Give the Pod a stable DNS record based on its name.
+	// - https://docs.k8s.io/concepts/workloads/controllers/statefulset/#stable-network-id
+	// - https://docs.k8s.io/concepts/services-networking/dns-pod-service/#pods
+	sts.Spec.ServiceName = naming.ClusterPodService(cluster).Name
+
+	// Set the StatefulSet update strategy to "RollingUpdate", and the Partition size for the
+	// update strategy to 0 (note that these are the defaults for a StatefulSet).
```

Comment:
🤔 I had to think for a moment about the partition "size" of zero. That made me think of an empty set and no-op. I think it's better to describe the effect of the values used.

```suggestion
	// Delete and recreate every pod of the StatefulSet when the pod template changes.
```

Should we also call out that known issue mentioned in the docs? Perhaps the link to the docs is enough?

> If you update the Pod template to a configuration that never becomes Running and Ready (for example, due to a bad binary or application-level configuration error), StatefulSet will stop the rollout and wait.

Reply:
Appended to the description to describe the effect, and also added a direct link to the "Forced rollback" section in the STS docs.